app/scripts/lib/util.js

```diff
@@ -144,6 +144,31 @@ function removeListeners (listeners, emitter) {
   })
 }
 
+function newFetch ({ timeout = 120000 } = {}) {
+  return function _fetch (url, opts) {
+    return new Promise(async (resolve, reject) => {
```

Comment:
The Promise constructor wrapper doesn't seem to serve any purpose here. Removing it would save a few cycles and result in better stack traces.